contracts/apps/AppStorage.sol

```diff
@@ -6,9 +6,10 @@ pragma solidity ^0.4.24;
 
 import "../common/UnstructuredStorage.sol";
 import "../kernel/IKernel.sol";
+import "./IAragonApp.sol";
 
 
-contract AppStorage {
+contract AppStorage is IAragonApp {
```

Comment:
Does it work if we enforce this in `AragonApp` directly?

Reply:
Nope, as `AppProxyBase` needs it. Actually, this was a big headache for my last commit (cc88e41), until I realized that `supportsInterface` was being shortcut and was not going through the delegated call because of this inheritance. :see_no_evil:

Reply:
But the `AppProxyBase` doesn't need to support the ERC165, I'm thinking about:

- Make `AragonApp` inherit from `IAragonApp`
- Leave the `AppStorage` contract as it is (without inheriting from `IAragonApp`
- Remove the forwarding for `supportsInterface` in `AppProxyBase`

Note that we want the AragonApp to support the `IAragonApp` interface, not the proxy. If the proxy does to shortcut its execution that's another part of the story I think. If we want to enforce the `kernel` and `appId` methods in the proxy as well, then I'd split the ERC165 part, otherwise any app proxy will be forced to that

Reply:
I tried to do that, but it didn’t compile because `kernel` and `appId` are defined in `AppStorage`. In particular it says: `Redeclaring an already implemented function as abstract`.

> otherwise any app proxy will be forced to that

I don’t fully get this. It will be the same in both cases: the proxy will redirect to the implementation when called `supportsInterface`.

Reply:
EDIT: I realize that changing the inheritance order, it works, so trying that solution again. It feels kind of weird to me that the contract implementing the functions can’t explicitly inherit the interface defining them though.